InternetStoreCHIReloaded/DAL/Repositories/UsersRepository.cs

```diff
@@ -0,0 +1,220 @@
+            }
+            else
+            {
+                user.UserCart.CartItems.Add(new ProductWithQuantityEntity()
+                {
+                    ProductId = productToCartDbModel.ProductId,
+                    Quantity = productToCartDbModel.Quantity,
+                });
+            }
+
+            bool saveSuccess = await SaveAsync();
+
+            var dbResponse = new DbResponse();
+
+            if (saveSuccess)
+            {
+                dbResponse.IsSuccessful = true;
+                dbResponse.Message = "Product was set successfully!";
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Nothing has changed.";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse> RemoveProductSetFromUserCartAsync(int userId, int productId)// removes entry from ProductWithQuantity table
+        {
+            var user = await GetUserWithCartAsync(userId);
+
+            DbResponse dbResponse = new DbResponse();
+
+            var productInCart = user.UserCart.CartItems.Where(p => p.ProductId == productId).FirstOrDefault();
+            if (productInCart != null)
+            {
+                user.UserCart.CartItems.Remove(productInCart);
+                _dbcontext.ProductsWithQuantity.Remove(productInCart);// also remove pwq entry
+
+                bool success = await UpdateAsync(user);
+
+                if (success)
+                {
+                    dbResponse.IsSuccessful = true;
+                    dbResponse.Message = "Product entry was removed successfuly!";
+                }
+                else
+                {
+                    dbResponse.IsSuccessful = false;
+                    dbResponse.Message = "Nothing has changed.";
+                }
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Product was not found in cart";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse<int>> GetQuantityOfProductInCartAsync(int userId, int productId)
+        {
+            var user = await GetUserWithCartAsync(userId);
+            var product = user.UserCart.CartItems.Where(p => p.ProductId == productId).FirstOrDefault();
+
+            var dbResponse = new DbResponse<int>();
+
+            if (product != null)
+            {
+                dbResponse.IsSuccessful = true;
+                dbResponse.Message = "Product quantity retrieval success!";
+                dbResponse.Data = product.Quantity;
+            }
+            else
+            {
+                dbResponse.IsSuccessful = false;
+                dbResponse.Message = "Product was not found in cart";
+            }
+
+            return dbResponse;
+        }
+
+        public async Task<DbResponse> MakeAnOrder(int userId)
```

Comment:
A lot business logic here. Please, move all business logic to BLL

Reply:
done